kazoo/recipe/lock.py

```diff
@@ -174,27 +174,17 @@ def acquire(self, blocking=True, timeout=None, ephemeral=True):
             The ephemeral option.
         """
 
-        def _acquire_lock():
-            got_it = self._lock.acquire(False)
-            if not got_it:
-                raise ForceRetryError()
-            return True
-
         retry = self._retry.copy()
         retry.deadline = timeout
 
         # Ensure we are locked so that we avoid multiple threads in
         # this acquistion routine at the same time...
-        locked = self._lock.acquire(False)
-        if not locked and not blocking:
+        thread_locked = self._thread_lock.acquire(
+            blocking=blocking, timeout=timeout if timeout is not None else -1
```

Comment:
I'm not a fan that python's threading chose to require timeout to be a number. Thankfully, it seems the other handler implementation's underlying Lock replacement also follows the python API, so this should be safe.

I could just replace the default parameter to -1 instead. Don't have a strong opinion either way

Reply:
IMO the way you implemented it is better: it does not change the method signature, so no chance to break someone's current code.